solver.rb

```diff
@@ -0,0 +1,36 @@
+class Solver
+  # Calculates the factorial of a non-negative integer
+  #
+  # @param num [Integer] The input integer
```

Comment:
- [Optional] Dear Team, your commit history does not clearly show you have written the tests before the solutions. It's important to write the individual tests separately and commit, then we can work on the solver for the tests. We need to show this in the commit history.